src/java/org/apache/cassandra/security/AbstractCryptoProvider.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.security;
+
+import java.security.Provider;
+import java.security.Security;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.cassandra.exceptions.ConfigurationException;
+
+import static java.lang.String.format;
+
+public abstract class AbstractCryptoProvider
+{
+    protected static final Logger logger = LoggerFactory.getLogger(AbstractCryptoProvider.class);
+    public static final String FAIL_ON_MISSING_PROVIDER_KEY = "fail_on_missing_provider";
```

Comment:
Seems like an overkill to me but why not.

Reply:
I don't insist, just have a concern if we are adding a crypto library from the classpath it could cause some problems for the tools/clients. Perhaps it is better not to override the flag itself, but to override the `class_name` with a system property, so that if something goes wrong we can disable the 'Amazon' dependency and reset it back to the default. Wdyt?

Reply:
but that can not be added here because it is "too late" (e.g. in the constructor). We would need to do that in DatabaseDescriptor because that is the place where this class is instantiated.

Reply:
@Mmuzaf btw why not to have both property for class as well as for that flag?

Reply:
I was thinking of a simple fix that would allow us to disable a new crypto provider without having to update the configuration file itself, which might not be very fast in a production environment that usually requires additional coordination between teams. Both flags also look OK too, but all that is up to you.

Reply:
added both